test/integration/controller/jobs/pod/pod_controller_test.go

```diff
@@ -323,6 +427,105 @@ var _ = ginkgo.Describe("Pod controller", ginkgo.Ordered, ginkgo.ContinueOnFailu
+				Queue("test-queue").
+				Obj()
+			pod2 := testingpod.MakePod("test-pod2", ns.Name).
+				Group("test-group").
+				GroupTotalCount("2").
+				Queue("test-queue").
+				Obj()
+			pod1LookupKey := client.ObjectKeyFromObject(pod1)
+			pod2LookupKey := client.ObjectKeyFromObject(pod2)
+
+			gomega.Expect(k8sClient.Create(ctx, pod1)).Should(gomega.Succeed())
+			gomega.Expect(k8sClient.Create(ctx, pod2)).Should(gomega.Succeed())
+
+			ginkgo.By("checking that workload is created for the pod group with the queue name")
+			wlLookupKey := types.NamespacedName{
+				Namespace: pod1.Namespace,
+				Name:      "test-group",
+			}
+			createdWorkload := &kueue.Workload{}
+			gomega.Eventually(func() error {
+				return k8sClient.Get(ctx, wlLookupKey, createdWorkload)
+			}, util.Timeout, util.Interval).Should(gomega.Succeed())
+
+			gomega.Expect(createdWorkload.Spec.PodSets).To(gomega.HaveLen(1))
+			gomega.Expect(createdWorkload.Spec.PodSets[0].Count).To(gomega.Equal(int32(2)))
+			gomega.Expect(createdWorkload.Spec.PodSets[0].Name).To(gomega.Equal("120fa2c0"))
+			gomega.Expect(createdWorkload.Spec.QueueName).To(gomega.Equal("test-queue"), "The Workload should have .spec.queueName set")
+
+			createdPod1 := &corev1.Pod{}
+			createdPod2 := &corev1.Pod{}
+			ginkgo.By("checking that all pods in group are unsuspended when workload is admitted", func() {
+				admission := testing.MakeAdmission(clusterQueue.Name, "120fa2c0").
+					Assignment(corev1.ResourceCPU, "default", "1").
+					AssignmentPodCount(createdWorkload.Spec.PodSets[0].Count).
+					Obj()
+				gomega.Expect(util.SetQuotaReservation(ctx, k8sClient, createdWorkload, admission)).Should(gomega.Succeed())
+				util.SyncAdmittedConditionForWorkloads(ctx, k8sClient, createdWorkload)
+
+				gomega.Eventually(func(g gomega.Gomega) []corev1.PodSchedulingGate {
+					g.Expect(k8sClient.Get(ctx, pod1LookupKey, createdPod1)).To(gomega.Succeed())
+					return createdPod1.Spec.SchedulingGates
+				}, util.Timeout, util.Interval).ShouldNot(
+					gomega.ContainElement(corev1.PodSchedulingGate{Name: "kueue.x-k8s.io/admission"}),
+				)
+				gomega.Eventually(func(g gomega.Gomega) []corev1.PodSchedulingGate {
+					g.Expect(k8sClient.Get(ctx, pod2LookupKey, createdPod2)).To(gomega.Succeed())
+					return createdPod2.Spec.SchedulingGates
+				}, util.Timeout, util.Interval).ShouldNot(
+					gomega.ContainElement(corev1.PodSchedulingGate{Name: "kueue.x-k8s.io/admission"}),
+				)
+
+				// TODO: check why this is not the last event, or maybe change it to check what we are interested in, a "Started" event was recorded (maybe for both pods)
+				// gomega.Eventually(func(g gomega.Gomega) bool {
+				// 	ok, err := testing.CheckLatestEvent(ctx, k8sClient, "Started", corev1.EventTypeNormal, fmt.Sprintf("Admitted by clusterQueue %v", clusterQueue.Name))
+				// 	g.Expect(err).NotTo(gomega.HaveOccurred())
+				// 	return ok
+				// }, util.Timeout, util.Interval).Should(gomega.BeTrue())
```

Comment:
If the previous tests namespace are properly deleted now, we could try to readable this check. Or just remove the commented code.